storey/flow.py

```diff
@@ -1191,15 +1196,199 @@ async def _load_schema(self, container, table_path):
+        condition_expression = ""
+
+        # Generating aggregation related expressions
+        if pending or not partitioned_by_key:
+            expressions, pending_updates = self._build_conditioned_feature_store_request(aggregation_element, pending)
+        # In case we get an indication the data is (probably) not updated from multiple workers (for example: pre sharded by key) run a
+        # simpler expression.
+        else:
+            expressions, pending_updates = self._build_simplified_feature_store_request(aggregation_element)
+            condition_expression = aggregation_element.storage_specific_cache.get(self._mtime_header_name, "")
+
+        # Generating additional cache expressions
+        if additional_data:
+            for name, value in additional_data.items():
+                expressions.append(f'{name}={self._convert_python_obj_to_expression_value(value)}')
+        update_expression = ';'.join(expressions)
+
+        return update_expression, condition_expression, pending_updates
+
+    def _build_conditioned_feature_store_request(self, aggregation_element, pending=None):
+        expressions = []
+
+        times_update_expressions = {}
+        pending_updates = {}
+        for name, bucket in aggregation_element.aggregation_buckets.items():
+            # Only save raw aggregates, not virtual
+            if bucket.should_persist:
+
+                if pending:
+                    items_to_update = pending[name]
+                else:
+                    pending_updates[name] = bucket.get_and_flush_pending()
+                    items_to_update = pending_updates[name]
+                for bucket_start_time, aggregation_value in items_to_update.items():
+                    # the relevant attribute out of the 2 feature attributes
+                    feature_attr = 'a' if int(bucket_start_time / bucket.window.max_window_millis) % 2 == 0 else 'b'
+                    array_attribute_name = f"{self._aggregation_attribute_prefix}{name}_{feature_attr}"
+                    array_time_attribute_name = f"{self._aggregation_time_attribute_prefix}{bucket.name}_{feature_attr}"
+
+                    expected_time = int(bucket_start_time / bucket.window.max_window_millis) * bucket.window.max_window_millis
+                    expected_time_expr = self._convert_python_obj_to_expression_value(datetime.fromtimestamp(expected_time / 1000))
+                    index_to_update = int((bucket_start_time - expected_time) / bucket.window.period_millis)
+
+                    get_array_time_expr = f"if_not_exists({array_time_attribute_name},0:0)"
+                    # TODO: Once Engine Expression bug is fixed remove occurrences of `tmp_arr` and `workaround_expression`
+                    workaround_expression = f';{array_attribute_name}=tmp_arr_{array_attribute_name};delete(tmp_arr_{array_attribute_name})'
```

Comment:
I know it's temporary, but looks like it shouldn't have a semicolon at the start.